0016-pre-shared-key/0016-pre-shared-key.md

```diff
@@ -0,0 +1,218 @@
+# Pre-Shared Key
+
+The Pre-Shared Key (PSK) protocol is an end-to-end transport protocol, used by the sender and receiver of an ILP payment to decide on a condition and fulfillment for a payment. The protocol also encrypts any additional data sent along with the payment, using AES-256-CTR. The PSK data is authenticated through an HMAC-SHA-256 which is used to generate the fulfillment of a PSK payment. Pseudocode for this protocol can be read [at the bottom of this spec](#pseudocode)
+
+As the name suggests, PSK relies on a pre-shared secret key. In application-layer protocols like SPSP, the receiver generates the shared key and shares it with the sender over an end-to-end secure communication channel such as HTTPS. 
+
+An advantage of PSK is that the pre-shared secret key only needs to be shared once. As long as both parties possess this key and are listening for transfers, they can send payments between one another.
+
+A disadvantage of PSK is that it is repudiable. Knowledge of a payment's fulfillment cannot be used as cryptographic proof that the receiver completed the payment, because both sender and receiver know how to generate the fulfillment with their pre-shared secret key. This causes no issues unless you have some other framework on top of ILP that relies on this cryptographic proof.
+
+## Flow
+
+0. The pre-shared secret key is shared out of band.
+0. The sender creates a random nonce, and calculates an HMAC of the nonce using the pre-shared secret key as the key. The result of this HMAC will be referred to as the payment encryption key.
+0. The sender constructs the PSK data:
+  0. The sender starts with the PSK status line: `PSK/1.0\n`
+  0. The sender appends the public headers (including the nonce), followed by `\n\n`.
```

Comment:
maybe link to the public headers section below, because this is the first time the headers concept is mentioned and it's a little unclear out of context what that means